### PR TITLE
Application default credentials for Stackdriver

### PR DESCRIPTION
**Pull request by ramonza**
Also added a convenience `Register` method to register as both a
stats and trace Exporter.

Fixes: #497

### CODE REVIEW

File: exporter/stackdriver/stackdriver.go [hunk 1]
```diff
@@ -73,9 +80,34 @@ type Exporter struct {
 	statsExporter *statsExporter
 }
 
+// Register sets up both stats and tracing exporting.
+func Register(o Options) (done func(), err error) {
```
Comment (odeke-em) [inline exporter/stackdriver/stackdriver.go]: Seems more like a convenience method that a user should make on their own than something we want to expose in our API, could we perhaps make it a helper for the test as you had done?

Comment (ramonza) [reply exporter/stackdriver/stackdriver.go]: My goal here was to make it as simple as possible for people to "turn on" OpenCensus for GCP client libraries. Adding this method makes this a one-liner as opposed to a 6-liner. @rakyll what do you think?

Comment (rakyll) [reply exporter/stackdriver/stackdriver.go]: I agree that done might be giving the wrong impression that it should be deferred right away. What about calling it unregister and document that it is more likely never be called but can be called to unregister and flush?

File: exporter/stackdriver/stackdriver.go [hunk 2]
```diff
@@ -73,9 +80,34 @@ type Exporter struct {
...
+		trace.UnregisterExporter(e)
+		e.Flush()
+	}, nil
+}
+
 // NewExporter creates a new Exporter that implements both stats.Exporter and
 // trace.Exporter.
 func NewExporter(o Options) (*Exporter, error) {
+	if o.ProjectID == "" {
+		creds, err := google.FindDefaultCredentials(context.Background(), traceapi.DefaultAuthScopes()...)
```
Comment (rakyll) [inline exporter/stackdriver/stackdriver.go]: Do we need to pass the auth scopes here? We are not using creds to create a new RoundTripper. We can omit it.

Comment (ramonza) [reply exporter/stackdriver/stackdriver.go]: true, but does it not make sense to defensively pass them anyway? it seems conceivable that someone might add a way to override default credentials by scope, in which case we want to pick up the ones associated with Stackdriver.

**Comment by ramonza** [comment]
@odeke-em removed the Register helper function, PTAL

**Pull request merged**